pkg/diff/diffbycommand.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"sigs.k8s.io/yaml"
+)
+
+// DiffByCommand converts the given objects into yaml and then runs the command to compare them.
+func DiffByCommand(command string, old, new interface{}) ([]byte, error) {
+	oldBytes, err := yaml.Marshal(old)
+	if err != nil {
+		return nil, err
+	}
+	newBytes, err := yaml.Marshal(new)
+	if err != nil {
+		return nil, err
+	}
+
+	return DiffBytesByCommand(command, oldBytes, newBytes)
+}
+
+// DiffBytesByCommand runs the command to compare the given bytes.
+func DiffBytesByCommand(command string, oldBytes, newBytes []byte) ([]byte, error) {
```

Comment:
What is the point of separate this function from `DiffByCommand`? 🤔

Reply:
I merged `DiffBytesByCommand()` into `DiffByCommand()`.

I used `DiffBytesByCommand()` from outside in previous commits, but it is not used now.